mr/src/test/java/org/apache/iceberg/mr/mapred/TestTableResolver.java

```diff
@@ -111,4 +111,21 @@ public void resolveTableFromPropertiesDefault() throws IOException {
     Assert.assertEquals(tableLocation.getAbsolutePath(), table.location());
   }
 
+  @Test(expected = UnsupportedOperationException.class)
+  public void resolveTableFromConfigurationHiveCatalog() throws IOException {
+    Configuration conf = new Configuration();
+    conf.set(InputFormatConfig.CATALOG_NAME, InputFormatConfig.HIVE_CATALOG);
+    conf.set(InputFormatConfig.TABLE_NAME, "table_a");
+
+    TableResolver.resolveTableFromConfiguration(conf);
+  }
+
+  @Test(expected = NullPointerException.class)
```

Comment:
We usually prefer using `AssertHelpers.assertThrows` here, but this is minor since you don't need to check that other state has not been modified after the failure.

Reply:
Since the above is almost certainly going to be refactored when we merge more logic between the two InputFormats we can tackle this then.

Reply:
I agree. I didn't want to block progress on Hive just for this.